rust/connlib/tunnel/src/peer.rs

```diff
@@ -268,10 +241,6 @@ impl ClientOnGateway {
         &self,
         packet: &MutableIpPacket<'_>,
     ) -> Result<(), connlib_shared::Error> {
-        if self.allowed_ips.longest_match(packet.source()).is_none() {
-            return Err(connlib_shared::Error::UnallowedPacket(packet.source()));
-        }
-
```

Comment:
@thomaseizinger  I was just thinking about this, and removing this means that another user, could use your ip as a source to send packets to a resource.

This means they could send something like a `RST` packet to kill your conneciton.

For this reason it might be better to still have the `allowed_ips`

Reply:
We could however implement this as just 2 `allowed_ipv4` and `allowed_ipv6`.

Which I might already use for #5049 to convert between ipv4 and ipv6 source ips

Reply:
> another user, could use your ip as a source to send packets to a resource.

Another user where? Is the packet coming out of the tunnel?

I'd agree that it is somewhat of a defense-in-depth thing. Technically, the source IP could be anything when coming out of the tunnel. But also, if somebody has access to the tunnel, they can also just mangle the IP to the one of the TUN device, no?

Reply:
Well, I mean this:

Say you have User A with public Key A  and IP A and user B with public key B and IP B.

Without this, user A could maliciously:

1. Mangle their encapsulated packet to have IP B
2. It's sent through the tunnel
3. Packet is successfully decapuselated using public key A
4. This associates the decapsulated packet to the conn-id of the corresponding public key A
5. We get the peer based on the conn-id
6. Now, w/o the `allowed_ips` the packets just gets forwarded to the resource as long as they have access to it
7. Resource believes the packet comes from IP B (sans nat-table)*

* IP A could make the source port collide with whatever user B is using and interrupt their connection or something else

If we have the `allowed_ips` step 6 would prevent always that user A uses anything but their assigned source ips since this is ensured by the portal

Reply:
Makes sense! I'll add a test for this to ensure we don't break it! :)